src/nativestore/DataPublisher.cpp

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright 2021 JasminGraph Team
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "./DataPublisher.h"
+
+#include "../server/JasmineGraphInstanceProtocol.h"
+#include "../util/logger/Logger.h"
+#include "../util/Utils.h"
+
+Logger data_publisher_logger;
+
+DataPublisher::DataPublisher(int worker_port, std::string worker_address) {
+    this->worker_port = worker_port;
+    this->worker_address = worker_address;
+    struct hostent *server;
+
+    server = gethostbyname(worker_address.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
```

Comment:
Lets use a logger instead of `std::cerr`

Reply:
Modified to print the worker address in the case of server is NULL.

Reply:
Ok